File: finalyearproject/repos/dkd-custody-ledger/internal/obs/tracing.go
package obs

import (
	"context"
	"crypto/rand"
	"encoding/hex"
)

type ctxKey string

const (
	CorrelationIDKey ctxKey = "correlation_id"
	TraceParentKey   ctxKey = "traceparent"
)

// NewTraceParent generates a W3C traceparent. The OpenTelemetry SDK is the integration point for
// span export; propagation works without it.
func NewTraceParent() string {
	tid := make([]byte, 16)
	sid := make([]byte, 8)
	_, _ = rand.Read(tid)
	_, _ = rand.Read(sid)
	return "00-" + hex.EncodeToString(tid) + "-" + hex.EncodeToString(sid) + "-01"
}

func CorrelationID(ctx context.Context) string {
	if v, ok := ctx.Value(CorrelationIDKey).(string); ok {
		return v
	}
	return ""
}

File: finalyearproject/repos/dkd-b2c-order/src/obs/tracing.ts
import { randomBytes } from "node:crypto";
// W3C traceparent propagation. The OpenTelemetry SDK is the export integration point.
export function newTraceParent(): string {
  return `00-${randomBytes(16).toString("hex")}-${randomBytes(8).toString("hex")}-01`;
}
export function newCorrelationId(): string { return randomBytes(16).toString("hex"); }
